src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -227,6 +226,7 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
         case GT_SUB_HI:
 #endif // !defined(TARGET_64BIT)
 
+        case GT_AND:
```

Comment:
Would it be possible to introduce the a new oper kind, such as ANDFLAGS, for this purpose? I think it is very unfortunate to overload this node in this way when we aren't doing it the containment way.

Reply:
Lowering would change `GT_AND` into `GT_ANDFLAGS` at the point it makes the determination that it is valuable to generate into flags instead and retypes it to `TYP_VOID`. So the if-conversion handling would keep using GT_AND.
`GT_ANDFLAGS` would essentially be a backend-only node. The hope would be that the current simple mental model for `GT_AND` could remain.

Reply:
Alternatively you could go back to your first approach with `GT_CCMP`, which I suppose would be sort of a generalization of `GT_ANDFLAGS`.

Reply:
Ok, so the only changes would be in lowering and code generation. I'll give this a try. Not sure yet how much it'd interfere with the `AND chain` code, but should be fine.

Reply:
Done. Still need to handle AND nodes being children of JTRUE nodes, for those cases where the chain needs to generate into a register.